app/webpacker/components/RegistrationsV2/Register/StripeWrapper.jsx

```diff
@@ -46,32 +41,29 @@ export default function StripeWrapper({
       }),
     );
   }, [connectedAccountId, stripePublishableKey]);
-  const handleDonation = useCallback(async (_, { value: newDonationAmount }) => {
-    const {
-      api_amounts: { stripe: stripeAmount },
-      human_amount: humanAmount,
-    } = await convertISOAmount(initialAmount + newDonationAmount, competitionInfo.currency_code);
-    setAmount(stripeAmount);
+
+  const handleDonation = useCallback(async (newDonationAmount) => {
     setDonationAmount(newDonationAmount);
-    setDisplayAmount(humanAmount);
-  }, [competitionInfo.currency_code, initialAmount]);
+    refetch();
+  }, [refetch]);
```

Comment:
I don't think a manual `refetch` trigger is necessary.
Just set the new donation amount using `setDonationAmount`. Doing this results in a new query key, which makes `useQuery` fetch new data (or retrieve information from the instance cache) automatically.